src/System.Linq/tests/ExceptTests.cs

```diff
@@ -125,5 +125,22 @@ public void ForcedToEnumeratorDoesntEnumerate()
             var en = iterator as IEnumerator<int>;
             Assert.False(en != null && en.MoveNext());
         }
+
+        [Fact]
+        public void HashSetWithBuiltInComparer_HashSetContainsNotUsed()
+        {
+            IEnumerable<string> input1 = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "a" };
+            IEnumerable<string> input2 = new[] { "A" };
```

Comment:
Why do both need to be hashsets?  The code below first uses input1.Except(input2), and then uses input2.Except(input1).